airflow/configuration.py

```diff
@@ -461,23 +461,53 @@ def _get_env_var_option(self, section: str, key: str):
 
     def _get_cmd_option(self, section: str, key: str):
         fallback_key = key + '_cmd'
-        # if this is a valid command key...
         if (section, key) in self.sensitive_config_values:
             if super().has_option(section, fallback_key):
                 command = super().get(section, fallback_key)
                 return run_command(command)
         return None
 
+    def _get_cmd_option_from_config_sources(
+        self, config_sources: ConfigSourcesType, section: str, key: str
+    ) -> Optional[str]:
+        fallback_key = key + '_cmd'
+        if (section, key) in self.sensitive_config_values:
+            section_dict = config_sources.get(section)
+            if section_dict is not None:
+                command_value = section_dict.get(fallback_key)
+                if command_value is not None:
+                    if isinstance(command_value, str):
+                        command = command_value
+                    else:
+                        command = command_value[0]
```

Comment:
In what cases does this end up as a list rather than a single string?

Reply:
It's a Tuple of [value, source] potentially - when ConfigSourceType is produced wuith "display_source=True", the section.get(key) will return a Tuple.

This is the main reasy why as prerequisite of this change I had to add Typing to conf: #23716  becuase it was very difficult to reason what types of values are returned where. 

So the change here is really not anything I "knew" about - it's more MyPy telling me that this might be either string or Tuple[str, str].

Reply:
BTW. Yeah. I would have done it differently actually and this whole conf requires rewriting at some point of time because the complexity of it grew over time with adding defaults, secrets, commands and finally deprecations which made it terribly complex overall - it took me good few hours to understand all the paths and understand how we should handle the deprecations to not fall into the trap of getting the "default" when there is a deprecation. 

So this whole part I think needs to be rewritten - but for 2.3.1 I think this is the minimum set of changes and very comprehensive test coverage covering all the tests cases I could possibly imagine so that we can actually safely cherry-pick that one to 2.3.1 (possibly).